lib/heex_formatter/algebra.ex

```diff
@@ -52,45 +30,61 @@ defmodule HeexFormatter.Algebra do
   end
 
   defp block_to_algebra([head | tail], context) do
-    initial =
+    {type, doc} =
       head
       |> to_algebra(context)
       |> maybe_force_unfit()
 
-    Enum.reduce(tail, initial, fn node, {prev_type, prev_doc} ->
+    Enum.reduce(tail, {head, type, doc}, fn next_node, {prev_node, prev_type, prev_doc} ->
       {next_type, next_doc} =
-        node
+        next_node
         |> to_algebra(context)
         |> maybe_force_unfit()
 
       cond do
         prev_type == :inline and next_type == :inline ->
-          {next_type, concat([prev_doc, flex_break(""), next_doc])}
+          on_break =
+            if next_doc != empty() and
+                 context.mode !== :script and
+                 context.mode !== :comment and
```

Comment:
Do we want to add a space during `pre`? We probably don't want to do that either... so maybe this should be:

```suggestion
                 context.mode == :normal
```